data/helpers.d/utils

```diff
@@ -28,6 +27,41 @@ ynh_exit_properly () {
 
 	ynh_print_err --message="!!\n  $app's script has encountered an error. Its execution was cancelled.\n!!"
 
+	# If the script is executed from the CLI, dump the end of the log that precedes the crash.
+	if [ "$YNH_INTERFACE" == "cli" ]
+	then
+		# Unset xtrace to not spoil the log
+		set +x
+
+		local ynh_log="/var/log/yunohost/yunohost-cli.log"
+
+		# Wait for the log to be fill with the data until the crash.
+		local timeout=0
+		while ! tail --lines=20 "$ynh_log" | grep --quiet "+ ynh_exit_properly"
+		do
+			((timeout++))
+			if [ $timeout -eq 500 ]; then
+				break
+			fi
+		done
+
+		echo -e "\e[34m\e[1mPlease find here an extract of the log before the crash:\e[0m" >&2
+		# Tail the last 30 lines of log of YunoHost
+		# But remove all lines after "ynh_exit_properly"
+		# Remove the timestamp at the beginning of the line
+		# Remove "yunohost.hook..."
+		# Add DEBUG and color it at the beginning of each log line.
+		echo -e "$(tail --lines=30 "$ynh_log" \
+			| sed '1,/+ ynh_exit_properly/!d' \
+			| sed 's/^[[:digit:]: ,-]*//g' \
+			| sed 's/ *yunohost.hook.*\]/ -/g' \
+			| sed 's/^WARNING /&/g' \
+			| sed 's/^DEBUG /&  /g' \
+			| sed 's/^INFO /&   /g' \
```

Comment:
So uh aren't the `DEBUG` supposed to be replaced ? (c.f. the screenshot)

Reply:
Those 3 lines, WARNING, DEBUG and INFO are only there to align the text after.
Those sed are only adding spaces.

Reply:
Alrighty, flagging as resolved ;P